R/col_types.R

```diff
@@ -364,8 +364,27 @@ col_spec_standardise <- function(file, col_names = TRUE, col_types = NULL,
   spec
 }
 
-guess_types <- function(datasource, tokenizer, locale, n = 1000) {
-  guess_types_(datasource, tokenizer, locale, n = n)
+
+check_guess_max <- function(guess_max, max_limit = .Machine$integer.max %/% 100) {
+  if (!(length(guess_max) == 1 && is.numeric(guess_max) &&
```

Comment:
How about this, the assert definition is slightly complicated because we want to evaluate the `...` lazily.

```r
assert <- function(msg, ...) {
  dots <- eval(substitute(alist(...)))
  for (x in dots) {
    if (!isTRUE(eval(x, parent.frame()))) {
      stop(msg, call. = FALSE)
    }
  }
  invisible(TRUE)
}

assert("`guess_max` must be a positive integer",
  length(guess_max) == 1, is.numeric(guess_max), is_integerish(guess_max),
  !is.na(guess_max), guess_max > 0)
```